prov/rxm/src/rxm_ep.c

```diff
@@ -512,13 +516,21 @@ static ssize_t rxm_ep_send_common(struct fid_ep *ep_fid, const struct iovec *iov
 		pkt_size = sizeof(*pkt) + pkt->hdr.size;
 	}
 
-	ret = fi_send(rxm_conn->msg_ep, pkt, pkt_size, desc_tx_buf, 0, tx_entry);
-	if (ret) {
-		FI_WARN(&rxm_prov, FI_LOG_EP_DATA, "fi_send for MSG provider failed\n");
-		goto err;
+	if (flags & FI_INJECT & ~FI_COMPLETION) {
+		ret = fi_inject(rxm_conn->msg_ep, pkt, pkt_size, 0);
```

Comment:
I'm not sure if I understand your question. FI_COMPLETION flag is usually used in conjunction with FI_SELECTIVE_COMPLETION which is not yet supported by rxm provider. Libfabric requires completions to be generated by default and that's the behavior right now.
> I guess it will be better to use fi_sendmsg with appropriate flags for all cases What do you think?

Can you elaborate?